### ISSUE TITLE
DevExpress Reports and ServiceProxyScript Internal error 500

### ISSUE THREAD

**Issue by sebitsi** [abpframework/abp#6208]
I'm using angular with WepApi (included Id4) latest (3.3.1) template.

When i register DevExpress reports into WebAPI pipeline ServiceProxyScript i got error 500.

Content in log is:

```
2020-11-15 21:43:32.885 +01:00 [INF] Route matched with {area = "Abp", action = "GetAll", controller = "AbpServiceProxyScript", page = ""}. Executing controller action with signature Microsoft.AspNetCore.Mvc.ActionResult GetAll(Volo.Abp.AspNetCore.Mvc.ProxyScripting.ServiceProxyGenerationModel) on controller Volo.Abp.AspNetCore.Mvc.ProxyScripting.AbpServiceProxyScriptController (Volo.Abp.AspNetCore.Mvc).
2020-11-15 21:43:32.887 +01:00 [WRN] Controller 'WebDocumentViewer' contains more than one action with name 'Invoke' for module 'app'. Ignored: System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] Invoke()
2020-11-15 21:43:32.887 +01:00 [WRN] Controller 'WebDocumentViewer' contains more than one action with name 'Invoke' for module 'app'. Ignored: System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] Invoke()
2020-11-15 21:43:32.887 +01:00 [INF] Executed action Volo.Abp.AspNetCore.Mvc.ProxyScripting.AbpServiceProxyScriptController.GetAll (Volo.Abp.AspNetCore.Mvc) in 2.1837ms
2020-11-15 21:43:32.887 +01:00 [INF] Executed endpoint 'Volo.Abp.AspNetCore.Mvc.ProxyScripting.AbpServiceProxyScriptController.GetAll (Volo.Abp.AspNetCore.Mvc)'

2020-11-15 21:43:32.908 +01:00 [ERR] An unhandled exception has occurred while executing the request.
System.ArgumentNullException: Value cannot be null. (Parameter 'httpMethod')
   at Volo.Abp.Check.NotNull[T](T value, String parameterName)
   at Volo.Abp.Http.Modeling.ActionApiDescriptionModel.Create(String uniqueName, MethodInfo method, String url, String httpMethod, IList`1 supportedVersions)
   at Volo.Abp.AspNetCore.Mvc.AspNetCoreApiDescriptionModelProvider.AddApiDescriptionToModel(ApiDescription apiDescription, ApplicationApiDescriptionModel applicationModel, ApplicationApiDescriptionModelRequestDto input)
   at Volo.Abp.AspNetCore.Mvc.AspNetCoreApiDescriptionModelProvider.CreateApiModel(ApplicationApiDescriptionModelRequestDto input)
   at Volo.Abp.Http.ProxyScripting.ProxyScriptManager.CreateScript(ProxyScriptingModel scriptingModel)
   at Volo.Abp.Http.ProxyScripting.ProxyScriptManager.<>c__DisplayClass6_0.<GetScript>b__0()

```
I know the problem exists because of two implementations of Invoke methods.

My first question is: 
Is it possible to replace Volo.Abp.AspNetCore.Mvc.AspNetCoreApiDescriptionModelProvider and how (where) i can do it ?

And second: 
What those "authetification failure messages" means:

```
2020-11-15 21:43:32.870 +01:00 [INF] Request starting HTTP/1.1 GET https://demo.xxx.xx/cws/Abp/ServiceProxyScript  
2020-11-15 21:43:32.879 +01:00 [INF] Identity.Application was not authenticated. Failure message: Unprotect ticket failed
2020-11-15 21:43:32.880 +01:00 [INF] Identity.Application was not authenticated. Failure message: Unprotect ticket failed
2020-11-15 21:43:32.880 +01:00 [INF] Identity.Application was not authenticated. Failure message: Unprotect ticket failed

```
Thanks for advance.

Edvin

**Comment by maliming** [comment]
hi, How to reproduce your problem?

**Comment by sebitsi** [comment]
@maliming There is nothing special.

Create new project based on ABP angular template.
Then follow [Dx Docs](https://docs.devexpress.com/XtraReports/400197/web-reporting/javascript-reporting/server-side-configuration/document-viewer/document-viewer-server-side-configuration-asp-net-core). When you run project first swagger fill fail, because of amigous methods names.

I can solve it using code like this:

```
services.AddSwaggerGen(c =>
{
    c.DocInclusionPredicate((docName, apiDesc) =>
    {
        if (!apiDesc.TryGetMethodInfo(out MethodInfo methodInfo)) return false;
        // Exclude all DevExpress reporting controllers
        return !methodInfo.DeclaringType.AssemblyQualifiedName.StartsWith("DevExpress", StringComparison.OrdinalIgnoreCase);
    });

```
But i don't know how "change"  Volo.Abp.AspNetCore.Mvc.AspNetCoreApiDescriptionModelProvider.
Or maybe is there any other way to exclude 3party code from ServiceProxyScript method ?

**Comment by maliming** [comment]
`ApiDescription's HttpMethod` is `string?`: **Gets or sets the supported HTTP method for this api, or null if all HTTP methods are supported.**

**Comment by sebitsi** [comment]
@maliming as i can see You are going to make httpMethod to be nullable.

Is there any way i can replace script provider or exlude some method /controlllers from scripting ?
We don't need all methods/controllers to be scripted.

**Comment by maliming** [comment]
hi @sebitsi 

I will consider adding a `selector` so that you can ignore some `ApiDescription`.

**Comment by sebitsi** [comment]
@maliming Thanks.
This will be very usefull. Regards.

**Comment by maliming** [comment]
New issue: https://github.com/abpframework/abp/issues/6251

**Comment by sebitsi** [comment]
@maliming  Great !! :) 
Is there possibility to implement it in 3.2.x roadmap  as patch ?
Moving to 4.x is huge step. .NET, angulat, Dx, etc., all have to be upgraded, tested before we can use it in production.

**Comment by maliming** [comment]
@sebitsi You can replace build-in `AspNetCoreApiDescriptionModelProvider` , See https://gist.github.com/maliming/5a76e6d65840530f60ab00e5b5d56c80#file-myaspnetcoreapidescriptionmodelprovider-cs-L59

**Comment by sebitsi** [comment]
@maliming Thanks for your gist. It solved my problem for now. 
I will close this issue.
